pkg/postgres/walker/schema.go

```diff
@@ -96,265 +122,110 @@ func (s *Schema) Print() {
-	var fields []Field
-	pksGrps := s.ParentKeysGroupedByTable()
-	for _, pks := range pksGrps {
-		fields = append(fields, pks.Fields...)
+func (s *Schema) getParentRelationship() SchemaRelationship {
+	rel := SchemaRelationship{
+		OtherSchema: s.Parent,
 	}
-	return fields
-}
-
-// ReferenceKeysGroupedByTable returns the keys from the referenced schemas that should be defined
-// as **foreign keys** for the current schema grouped by parent schema.
-func (s *Schema) ReferenceKeysGroupedByTable() []TableFieldsGroup {
-	pks := make([]TableFieldsGroup, 0, len(s.ReferencedSchema))
-	// Find all the local fields that are already defined as foreign keys.
-	localFKs := s.localFKs()
-
-	for _, parent := range s.ReferencedSchema {
-		currPks := parent.ResolvedPrimaryKeys()
-		filtered := make([]Field, 0, len(currPks))
-		for idx := range currPks {
-			pk := &currPks[idx]
-			field, found := localFK(pk, localFKs)
-			if !found {
-				tryParentify(pk, parent)
-				filtered = append(filtered, *pk)
-				continue
-			}
-
-			// If the referenced parent field is already embedded in child but without constraint, skip it.
-			if field.Options.Reference != nil && field.Options.Reference.NoConstraint {
-				continue
-			}
-			// If the referenced parent field is already embedded in child, use the child field names.
-			pk.Name = field.Name
-			pk.Reference = pk.ColumnName
-			pk.ColumnName = field.ColumnName
-			pk.Options.Reference = field.Options.Reference
-			filtered = append(filtered, *pk)
-		}
-
-		if len(filtered) > 0 {
-			pks = append(pks, TableFieldsGroup{Table: parent.Table, Fields: filtered})
+	for _, f := range s.Fields {
+		if ref := f.Options.Reference; ref != nil && ref.OtherSchema == s.Parent {
+			rel.MappedColumnNames = append(rel.MappedColumnNames, ColumnNamePair{
+				ColumnNameInThisSchema:  f.ColumnName,
+				ColumnNameInOtherSchema: ref.ColumnName,
+			})
 		}
 	}
-	return pks
+	return rel
 }
 
-// ParentKeysGroupedByTable returns the keys from the (proto-)parent schemas that should be defined
-// as **foreign keys** for the current schema grouped by parent schema.
-func (s *Schema) ParentKeysGroupedByTable() []TableFieldsGroup {
-	pks := make([]TableFieldsGroup, 0, len(s.ReferencedSchema))
-	// Find all the local fields that are already defined as foreign keys.
-	localFKs := s.localFKs()
-
-	for _, parent := range s.ReferencedSchema {
-		currPks := parent.ResolvedPrimaryKeys()
-		filtered := make([]Field, 0, len(currPks))
-		for idx := range currPks {
-			pk := &currPks[idx]
-			if _, found := localFK(pk, localFKs); found {
-				continue
-			}
-			tryParentify(pk, parent)
-			filtered = append(filtered, *pk)
-		}
-
-		if len(filtered) > 0 {
-			pks = append(pks, TableFieldsGroup{Table: parent.Table, Fields: filtered})
+// AllRelationships returns all SchemaRelationships this schema has.
+// It includes relationships to everything -- schemas this schema references, other schemas that
+// reference this schema, parent and children.
```

Comment:
irrespective of referential constraint exists?

Reply:
Correct. I'll update the comment to be explicit.